setup.md

```diff
@@ -0,0 +1,92 @@
+### Install mbed-cli to a venv
+
+Install mbed-cli to a virtual environment, so that all the things mbed-cli
+installs automatically don't interfere with other things you have installed as
+part of your main environment (which may be managed by your package manager,
+etc.).
+
+```sh
+virtualenv venvs/mbed
+source ~/venvs/mbed/bin/activate
+pip install mbed-cli
+```
+
+### Install the GCC_ARM toolchain
+
+We'll install arm-none-eabi-gcc to `/opt`. You may want to use `~/.local` or
+some other place. Where doesn't matter too much, so long as its in your `$PATH`.
+
+```sh
+wget -q https://developer.arm.com/-/media/Files/downloads/gnu-rm/5_4-2016q3/gcc-arm-none-eabi-5_4-2016q3-20160926-linux.tar.bz2 -O gcc-arm-none-eabi-5_4-2016q3-20160926-linux.tar.bz2 && tar -xjf gcc-arm-none-eabi-5_4-2016q3-20160926-linux.tar.bz2 -C /opt
+rm gcc-arm-none-eabi-5_4-2016q3-20160926-linux.tar.bz2
+```
+
+Remember to add `/opt/gcc-arm-none-eabi-5.4-2016q3/bin` to your `$PATH`.
+
+### Install the ATECC608A example
+
+```sh
+git clone git@github.com:ARMmbed/mbed-os-example-atecc608a.git
+cd mbed-os-example-atecc608a/
+source ~/venvs/mbed/bin/activate
+mbed new .
+mbed deploy
+mbed compile -t GCC_ARM -m K64F --flash --sterm
+```
+
+### PSA Crypto driver development
+
+Add any driver files (none created so far) to the mbed-os-atecc608a folder. It
+is backed by the
+[mbed-os-atecc608a](https://github.com/ARMmbed/mbed-os-atecc608a/tree/mbed-cryptoauthlib)
+repo, which any application wanting to use Mbed OS and the ATECC608A will
+depend on (same as how this example depends on it).
+
+Any Mbed Crypto changes needed to support your driver work should be done in
+`mbed-os/features/mbedtls/mbed-crypto/importer/TARGET_IGNORE/mbed-crypto`, and copied
+into Mbed OS's idiosyncratic file layout before use using the importer.
+
+For the first-time setup only, if you need to make any Mbed Crypto changes,
+perform the following operations:
+
+```sh
+cd mbed-os/features/mbedtls/mbed-crypto/importer
+make update
+```
+
+For every change you make in Mbed Crypto (which you'd make inside
+`mbed-os/features/mbedtls/mbed-crypto/importer/TARGET_IGNORE/mbed-crypto`),
+update Mbed OS with those changes by performing the following operations.
+
+```sh
+curdir=`pwd`
+cd mbed-os/features/mbedtls/mbed-crypto/importer
+make
+cd $curdir
+```
+
+If you don't like having multiple copies of mbed-crypto or mbed-os lying
+around, feel free to make symlinks or git worktrees as desired.
```

Comment:
Last I tried, mbed didn't like git worktrees, and git worktrees are largely incompatible with submodules.

Reply:
It works for me and how I do things. You can make up your own mind if you want to do the same, which is why there is "feel free" reasoning.

Reply:
I feel like this tip is not intrusive, and properly noted as a non-obligatory step, so I'd leave it as is.

Reply:
It didn't work for me when I tried it. @Patater I'd like to know how you do it!

Reply:
I did `mbed deploy` once. Then replaced repos with symlinks. Running subsequent `mbed deploy` will update the symlinked repo.

`mbed deploy` doesn't know what a worktree is and won't like being symlinked to one of those. It also doesn't like having a real worktree within the project folder (non-symlinked). However, `mbed compile` works file with worktrees. You can run `git` commands to update tot he latest lib file, if you want. It'd be nice if mbed recognized worktrees; then we could have worktrees inside every example.